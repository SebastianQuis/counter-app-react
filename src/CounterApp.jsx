import React, { useState } from 'react'

import PropTypes from 'prop-types'


export const CounterApp = ( {value} ) => {
    console.log('render!');
    // const {value} = props;
    const [counter, setCounter] = useState(value);

    const sumar = () => {
        // setCounter(counter + 1);
        setCounter( (c) => c+1)
    }

    const restar = () => {
        (counter < 1) 
            ? setCounter( 0 ) 
            : setCounter( counter - 1);
    }

    const resetear = () =>  setCounter( value );
    

    return (
        <>
            <h1>CounterApp</h1>
            <p> {counter} </p>
            <button onClick={ sumar }> +1 </button>
            <button onClick={ restar }> -1 </button>
            <button onClick={ resetear }> Reset </button>
        </>
    )
}


CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
