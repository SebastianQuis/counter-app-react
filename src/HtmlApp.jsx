import React from 'react'

import PropTypes from 'prop-types'

const nombre = 'Sebastian';
const user = {
    nombre: 'Sebasxd',
    apodo: 'El pepe',
    direccion: {
        calle: 'Junin',
        num: 150
    }
}

// CODIGO QUE FUNCIONA SOLO UNA VEZ 
const getNombre = () => `${user.nombre}`;
const numArr = [1,2,3,4,5,6,7]

export const HtmlApp = ({nombre, apodo, edad}) => {
    // CODIGO QUE FUNCIONA CADA VEZ QUE ACTUALICE
    // STATEFULL
    // console.log(props); // viene del componente padre
    return (
        <>
            <h1>Impresion de variables en HTML</h1>
            <h2> {nombre} </h2>
            <p> {apodo} </p>
            <p> {edad + 5} </p>
            {/* <code>{ JSON.stringify( user ) }</code>
            <p>
                { numArr.map( num => {
                    return <p> {num}: Seb </p>
                }) }
            </p>
            <p>{getNombre()}</p> */}
        </>
    )
}

// TODO - TIPOS DE DATOS PARA PARAMETROS DEL COMPONENETE
HtmlApp.propTypes = {
    nombre: PropTypes.string.isRequired,
    edad: PropTypes.number.isRequired,
}

// TODO - definir los valores de los proptypes.
HtmlApp.defaultProps = {
    apodo: 'Sin apodo',
    edad: 1,
    nombre: 'Sin nombre',
}



