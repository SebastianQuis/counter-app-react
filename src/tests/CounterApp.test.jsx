import { fireEvent, getByText, render, screen } from "@testing-library/react"
import { HtmlApp } from "../HtmlApp";
import { CounterApp } from "../CounterApp";


describe('Pruebas en <CounterApp/>', () => { 
     const initValue = 100;

     test('should have the snapshot', () => { 
          const { container } = render( 
               <CounterApp  value={initValue}/> 
          );
          expect( container ).toMatchSnapshot(); // crea archivo en __snapshot__
     })

     test('should number: 100', () => { 
          render( <CounterApp  value={initValue}/> );
          // expect( screen.getByText(initValue).innerHTML ).toContain("100");
          expect( screen.getByText(initValue) ).toBeTruthy();
     })
     
     test('should add 1 in value: 101', () => { 
          render( <CounterApp  value={initValue}/> );
          fireEvent.click( screen.getByText('+1') );
          screen.debug();
          expect( screen.getByText( initValue + 1 ).innerHTML ).toBe( `${initValue + 1}` );
          // expect( screen.getByText( value ).innerHTML ).toBeTruthy();
     });
     
     test('should add 1 in value: 100', () => { 
          render( <CounterApp  value={initValue}/> );
          fireEvent.click( screen.getByText('-1') );
          expect( screen.getByText( initValue - 1 ).innerHTML ).toBe( `${initValue - 1}` );
     });
     
     
     test('should reset the number to: 100', () => { 
          render( <CounterApp  value={initValue}/> );
          fireEvent.click( screen.getByText('+1') );
          fireEvent.click( screen.getByText('+1') );
          
          fireEvent.click( screen.getByRole('button', { name: 'btn-reset' }) );

          expect( screen.getByText(initValue).innerHTML ).toBe('100')
      })

     
})


