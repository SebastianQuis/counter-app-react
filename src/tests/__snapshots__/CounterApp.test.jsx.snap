// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp/> should have the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <p>
    100
  </p>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
